k6error/internal.go

```diff
@@ -0,0 +1,42 @@
+package k6error
+
+import (
+	"errors"
+	"fmt"
+)
+
+// Internal represents an internal error which
+// should abort the whole test run. These sorts
+// of errors should be used when we want to
+// signal to the maintainer of the project that
+// something unexpected has happened.
+type Internal struct{ err error }
+
+// AsInternal wraps the given error into an
+// Internal error type.
+func AsInternal(err error) *Internal { return &Internal{err: fmt.Errorf("internal: %w", err)} }
+
+// AsInternal wraps the given msg into an
+// Internal error type.
+func AsInternalError(msg string) *Internal {
+	return &Internal{err: fmt.Errorf("internal: %w", errors.New(msg))}
+}
+
+// Error will return the msg of the wrapped error.
+func (i *Internal) Error() string { return i.err.Error() }
+
+// Internal will return true to signal that it
+// is an internal error.
+func (i *Internal) Internal() bool { return true }
```

Comment:
Do we need this? Can't we work with std's `errors.Is()` ? Also in order to call this method wouldn't we need to cast an `error` into our `Internal` (maybe through `errors.As()`) ?

Reply:
I have the same question [here](https://github.com/grafana/xk6-browser/pull/744#issuecomment-1411925677). Take a look if you haven't seen it :)

Reply:
Yeah good point, we don't need this 🙂

I've found it to be useful when working with `temporary` errors (mainly for backend services) since there are other error types within go that implement `Temporary()`.